Chapter9/Chapter9/Controller/RestaurantDetailViewController.swift

```diff
@@ -30,11 +42,28 @@ class RestaurantDetailViewController: UIViewController {
         customizeNavigationBar()
         setupInfoCell()
     }
+
+    private func setRate() {
+
+            let scaleTransform = CGAffineTransform(scaleX: 0.1, y: 0.1)
+            self.restaurantHeaderView.ratingImageView.transform = scaleTransform
+            self.restaurantHeaderView.ratingImageView.alpha = 0
+
+            UIView.animate(withDuration: 0.4, delay: 0, usingSpringWithDamping: 0.3, initialSpringVelocity: 0.7, options: [], animations: {
+                self.restaurantHeaderView.ratingImageView.transform = .identity
+                self.restaurantHeaderView.ratingImageView.alpha = 1
+            }, completion: nil)
+
+    }
     override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
         if segue.identifier == "showMap" {
             let destinationController = segue.destination as! MapViewController
             destinationController.restaurant = restaurantDetails!
         }
+        if segue.identifier == "showReview" {
+            let destinationController = segue.destination as! ReviewViewController
+            destinationController.restaurant = restaurantDetails!
```

Comment:
You should use a guard for the restaurant, you don't know for sure it has a value